module/VuFind/src/VuFind/Controller/OverdriveController.php

```diff
@@ -308,11 +293,290 @@ public function holdAction()
                 'cover',
                 'title',
                 'actionTitleCode',
-                'listAuthors'
+                'listAuthors',
+                'edition'
             )
         );
 
         $view->setTemplate('RecordDriver/SolrOverdrive/hold');
         return $view;
     }
+
+    /**
+     * Hold Confirm Result
+     *
+     * Get result of the action
+     *
+     * @return obj Result Object
+     */
+    public function getHoldConfirmRes()
+    {
+        $od_id = $this->params()->fromQuery('od_id');
+        $result = $this->connector->getResultObject();
+        // Check to make sure they don't already have this checked out.
+        // Shouldn't need to refresh.
+        if ($checkout = $this->connector->getCheckout($od_id, true)) {
+            $result->status = false;
+            $result->data = (object)[];
+            $result->data->checkout = $checkout;
+            $result->code = 'OD_CODE_ALREADY_CHECKED_OUT';
+            $this->debug("title already checked out: $od_id");
+        } elseif ($hold = $this->connector->getHold($od_id, true)) {
+            $result->status = false;
+            $result->data = (object)[];
+            $result->data->hold = $hold;
+            $result->code = 'OD_CODE_ALREADY_ON_HOLD';
+            $this->debug("title already on hold: $od_id");
+        } else {
+            $result->status = true;
+        }
+        return $result;
+    }
+
+    /**
+     * Confirm Checkout Result
+     *
+     * Get result of the action
+     *
+     * @return obj Result Object
+     */
+    public function getConfirmCheckoutRes()
+    {
+        $od_id = $this->params()->fromQuery('od_id');
+        $rec_id = $this->params()->fromQuery('rec_id');
+        // Load the Record Driver. Should be a SolrOverdrive driver.
+        $driver = $this->serviceLocator->get(\VuFind\Record\Loader::class)->load(
+            $rec_id
+        );
+        $formats = $driver->getDigitalFormats();
+        // Looks like this is a magazine...
+        if (current($formats)->id == 'magazine-overdrive') {
```

Comment:
Is this a safe enough check? Is there always going to be exactly one digital format? If not, would it be better to create a support method to search the whole array?

Reply:
I think it is. Ebooks and audiobooks can have multiple formats but magazines just come in one format